h/static/scripts/controllers/character-limit-controller.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+function CharacterLimitController(element) {
+  var maxlength = parseInt(element.getAttribute('data-maxlength'));
+  var counter = element.parentElement.querySelector('.js-character-counter');
```

Comment:
Controllers should avoid referencing elements outside of the sub-tree that they are upgrading. I would suggest fixing this by adding the controller on a class in `mapping_item.jinja2` which defines the container for the `<textarea>`.